### ISSUE TITLE
Operations with mismatched dimensions

### ISSUE THREAD

**Issue by lbilli** [JuliaMatrices/BandedMatrices.jl#118]
I was surprised to see that these operations succeed even though dimensions don't match:
```julia
julia> using BandedMatrices

julia> m = BandedMatrix(Eye(3), (0,0))
3×3 BandedMatrix{Float64,Array{Float64,2},Base.OneTo{Int64}}:
 1.0   ⋅    ⋅ 
  ⋅   1.0   ⋅ 
  ⋅    ⋅   1.0

julia> m * [1,2]
3-element Array{Float64,1}:
 1.0
 2.0
 0.0

julia> m * [1, 2, 3, 4]
3-element Array{Float64,1}:
 1.0
 2.0
 3.0

julia> m \ [1, 2]
3-element Array{Float64,1}:
 1.0                  
 2.0                  
 6.92307382816255e-310
```
Needless to say, if a regular `Matrix` is used instead, `DimensionMismatch` exceptions are thrown.

Is this the expected behavior?

**Comment by dlfivefifty** [comment]
This is definitely not expected

**Comment by lbilli** [comment]
Fixed in v0.10.0

**Comment by dlfivefifty** [comment]
I'm going to reopen this to remember to add tests.
